examples/sha256.rs

```diff
@@ -94,33 +107,30 @@ impl<F: LurkField> Coprocessor<F> for Sha256Coprocessor<F> {
         hasher.update(input);
         let result = hasher.finalize();
 
-        let u: Vec<u128> = (0..2)
+        let mut u: Vec<u128> = (0..2)
             .map(|i| {
                 let a: [u8; 16] = result[(16 * i)..(16 * (i + 1))].try_into().unwrap();
                 u128::from_be_bytes(a)
             })
             .collect();
+        
+        u.reverse();
+
+        assert_eq!(u, self.expected);
 
-        let blah = &[u[0], u[1]].map(|x| s.intern_num(u128_into_scalar::<F>(x)));
-        s.list(blah)
+        s.get_t()
```

Comment:
I'm not 100% sure of the intended interface, but this is suspicious. You're returning `t` which makes this feel like a predicate, but you're panicking in the false case. I think you probably want to either return a Lurk boolean that is true when the result is as expected and false otherwise. In that case, you can use `Store::as_lurk_boolean()`.

Reply:
Hmm, looking back at the circuit, I see that this is consistent — but it's a bit weird. For this 'benchmark', it's probably fine, although what I described makes a bit more sense as an API (in my opinion). It would cost a few constraints, but in this context that's not significant. Consider whether you want to implement it as an exercise in coprocessor engineering.